docking.js

```diff
@@ -2017,9 +2017,17 @@ var DockManager = class DashToDock_DockManager {
                     break;
             }
 
-            const mainDockProperties = {};
-            if (dock === this.mainDock)
-                mainDockProperties.onComplete = callback;
+            let newCallback = () => {};
+            if (dock === this.mainDock) {
+                let destroyId = 0;
+                newCallback = () => {
+                    if (destroyId)
+                        dash.disconnect(destroyId);
+
+                    callback();
+                };
+                destroyId = dash.connect('destroy', newCallback);
+            }
```

Comment:
I'd prefer to keep using `mainDockProperties`, so can be extended in future, so here:

```suggestion
            const mainDockProperties = {}
            if (dock === this.mainDock) {
                const destroyId = dash.connect('destroy', () =>
                    mainDockProperties.onStopped());
                mainDockProperties.onStopped = () => {
                    dash.disconnect(destroyId);
                    callback();
                };
            }
```

Reply:
I wouldn't call `mainDockProperties` a cleanup. It's code that doesn't need to exist and is misleading because it's being passed in to all docks, not just the main one. Only its value changes. So the name is wrong at least: `s/mainDockProperties/dockProperties/`.

Reply:
Done.